Fix strategies bug in pandas==1.4.0

**Describe the bug**

The strategies module errors out for pandas==1.4.0 for the pandas-native string dtype, see [here](https://github.com/pandera-dev/pandera/runs/4914520273?check_suite_focus=true) for an example.

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of pandera.
- [ ] (optional) I have confirmed this bug exists on the master branch of pandera.

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Error

```python
self = <StringArray>
['', '', '', '', nan]
Length: 5, dtype: string

    def _validate(self):
        """Validate that we only store NA or strings."""
        if len(self._ndarray) and not lib.is_string_array(self._ndarray, skipna=True):
>           raise ValueError("StringArray requires a sequence of strings or pandas.NA")
E           ValueError: StringArray requires a sequence of strings or pandas.NA
```

#### Expected behavior

There shouldn't be an error when synthesizing dataframes with a schema that uses the pandas string dtype.

#### Desktop (please complete the following information):

 - OS: [e.g. iOS]
 - Browser [e.g. chrome, safari]
 - Version [e.g. 22]

--- a/pandera/strategies.py
+++ b/pandera/strategies.py
@@ -74,6 +74,8 @@ def _mask(
 ) -> Union[pd.Series, pd.Index]:
     if pd.api.types.is_timedelta64_dtype(val):
         return val.mask(null_mask, pd.NaT)
+    elif val.dtype == pd.StringDtype():
+        return val.mask(null_mask, pd.NA)
     return val.mask(null_mask)
 
 

--- a/pandera/typing/pandas.py
+++ b/pandera/typing/pandas.py
@@ -129,6 +129,13 @@ class DataFrame(DataFrameBase, pd.DataFrame, Generic[T]):
         out = writer(*args, **(config.to_format_options or {}))
         if buffer is None:
             return out
+        elif buffer.closed:
+            raise IOError(
+                f"pandas=={pd.__version__} closed the buffer automatically "
+                f"using the serialization method {writer}. Use a later "
+                "version of pandas or use a different the serialization "
+                "format."
+            )
         buffer.seek(0)
         return buffer
 

--- a/tests/core/test_from_to_format_conversions.py
+++ b/tests/core/test_from_to_format_conversions.py
@@ -125,6 +125,12 @@ def test_from_format(schema, to_fn, buf_cls):
         buf = None if buf_cls is None else buf_cls()
         arg = to_fn(df, *([buf] if buf else []))
         if buf:
+            if buf.closed:
+                pytest.skip(
+                    "skip test for older pandas versions where to_pickle "
+                    "closes user-provided buffers: "
+                    "https://github.com/pandas-dev/pandas/issues/35679"
+                )
             buf.seek(0)
             arg = buf
         if invalid:
@@ -164,7 +170,14 @@ def test_to_format(schema, from_fn, buf_cls):
         return df
 
     df = mock_dataframe()
-    out = fn(df)
+    try:
+        out = fn(df)
+    except IOError:
+        pytest.skip(
+            f"pandas=={pd.__version__} automatically closes the buffer, for "
+            "more details see: "
+            "https://github.com/pandas-dev/pandas/issues/35679"
+        )
     if buf_cls and not isinstance(out, buf_cls):
         out = buf_cls(out)
     out_df = from_fn(out)

--- a/tests/koalas/test_schemas_on_koalas.py
+++ b/tests/koalas/test_schemas_on_koalas.py
@@ -293,7 +293,15 @@ def test_nullable(
                 ks.DataFrame(nonnull_sample)
             return
     else:
-        ks_null_sample = ks.DataFrame(null_sample)
+        try:
+            ks_null_sample = ks.DataFrame(null_sample)
+        except TypeError as exc:
+            if "can not accept object <NA> in type" not in exc.args[0]:
+                raise
+            pytest.skip(
+                "koalas cannot handle native pd.NA type with dtype "
+                f"{dtype.type}"
+            )
         ks_nonnull_sample = ks.DataFrame(nonnull_sample)
         n_nulls = ks_null_sample.isna().sum().item()
         assert ks_nonnull_sample.notna().all().item()

